jobs/build/aws-ami/Jenkinsfile

```diff
@@ -0,0 +1,117 @@
+                                 [$class: 'hudson.model.ChoiceParameterDefinition',
+                                    choices: "openshift-enterprise\norigin",
+                                    defaultValue: 'openshift-enterprise',
+                                    description: '''origin                    Openshift origin  <br>
+                                                    openshift-enterprise      Openshift Enterpriseonline <br>
+                                    ''',
+                                    name: 'DEPLOYMENT_TYPE'],
+                                 [$class: 'hudson.model.StringParameterDefinition', defaultValue: 'opstest', description: 'Clusterid to set for ami build. Also used for VPC and other settings', name: 'CLUSTERID'],
+                                 [$class: 'hudson.model.StringParameterDefinition', defaultValue: '3.7.0', description: 'Release version (matches version in branch name for release builds)', name: 'RELEASE_VERSION'],
+                                 [$class: 'BooleanParameterDefinition', defaultValue: false, description: 'Mock run to pickup new Jenkins parameters?.', name: 'MOCK'],
+                                 [$class: 'hudson.model.StringParameterDefinition', defaultValue: 'ami-068a787c', description: 'Base AMI id to build from.', name: 'BASE_AMI_ID'],
+                                 [$class: 'hudson.model.StringParameterDefinition', defaultValue: 'ami_base', description: 'Base AMI instance name.', name: 'BASE_AMI_NAME'],
+                                 [$class: 'hudson.model.StringParameterDefinition', defaultValue: 'ami_builder_key', description: 'Name of the AWS SSH key user to use.', name: 'AWS_SSH_KEY_USER'],
+                                 [$class: 'hudson.model.StringParameterDefinition', defaultValue: 'https://mirror.openshift.com/enterprise/online-int/latest/x86_64/os/', description: 'Base url for repository.', name: 'YUM_BASE_URL'],
+                         ]
+                ],
+        ]
+)
+
+// Force Jenkins to fail early if this is the first time this job has been run/and or new parameters have not been discovered.
+echo "MAIL_LIST_SUCCESS:[${MAIL_LIST_SUCCESS}], MAIL_LIST_FAILURE:[${MAIL_LIST_FAILURE}], FORCE_REBUILD:${FORCE_REBUILD}, CLUSTERID:${CLUSTERID}, RELEASE_VERSION:${RELEASE_VERSION}, BASE_AMI_ID:${BASE_AMI_ID}, BASE_AMI_NAME:${BASE_AMI_NAME}, AWS_SSH_KEY_USER:${AWS_SSH_KEY_USER}, YUM_BASE_URL:${YUM_BASE_URL}"
+
+if ( MOCK.toBoolean() ) {
+    error( "Ran in mock mode to pick up any new parameters" )
+}
+
+node {
+
+    set_workspace()
+    
+    stage('build') {
+        try {
+
+            sh(script: '''if [ -d "${WORKSPACE}/openshift-ansible" ]; then 
+                cd ${WORKSPACE}/openshift-ansible ; 
+                git pull; 
+            else
+                git clone https://github.com/openshift/openshift-ansible
+            fi
+            
+            cd ${WORKSPACE}
```

Comment:
This will have no effect. use dir( .. ) to ensure pipeline steps run where you intend them to run: https://github.com/openshift/aos-cd-jobs/blob/master/jobs/build/ocp/Jenkinsfile#L152

Reply:
This is called inside of a sh().  Is there a way to do the surrounding logic in groovy?  I can remove the `cd ${WORKSPACE}` if desired but I will still required git cloning if the workspace does not exist.

Any help here (an example) would be appreciated.